src/StoreApi/Routes/Checkout.php

```diff
@@ -177,6 +177,14 @@ protected function get_route_post_response( \WP_REST_Request $request ) {
 		// Persist customer address data to account.
 		$order_controller->sync_customer_data_with_order( $order_object );
 
+		/*
+		* Fire woocommerce_blocks_checkout_order_processed, should work the same way as woocommerce_checkout_order_processed
+		* But we're opting for a new action because the original ones attaches POST data.
+		* NOTE: this hook is still experimental, and might change or get removed.
+		* @todo: Document and stabilize __experimental_woocommerce_blocks_checkout_order_processed
+		*/
+		do_action( '__experimental_woocommerce_blocks_checkout_order_processed', $order_object->get_id(), $order_object );
```

Comment:
Do we need to pass the order ID and the order object? Surely the ID is redundant if we are passing the whole object.

Reply:
That's a good idea, I removed the ID.
Initially, I was trying to keep parity with the original hook.
Using the object only would require some refactoring in subscriptions to create a wrapper function.

Reply:
One other risk is passing outdated data or data that is yet to be saved, but we should have saved everything before this point.